xdsl/dialects/mod_arith.py

```diff
@@ -0,0 +1,40 @@
+from abc import ABC
+
+from xdsl.dialects.arith import signlessIntegerLike
+from xdsl.ir import Attribute, Dialect, SSAValue
+from xdsl.irdl import IRDLOperation, irdl_op_definition, operand_def, result_def
+from xdsl.traits import Pure
+
+
+class ModArithOp(IRDLOperation, ABC):
+    # assemblyFormat = "operands attr-dict `:` functional-type(operands, results)"
+    pass
+
+
+class BinaryOp(ModArithOp):
+    lhs = operand_def(signlessIntegerLike)
+    rhs = operand_def(signlessIntegerLike)
+    output = result_def(signlessIntegerLike)
+    # assemblyFormat ="$lhs $rhs attr-dict `:` type($output)"  # operands directive not in asm format
```

Comment:
What does this comment mean? Does the assembly format not work?

Reply:
Hmm, I think the problem is that the magic fairy dust that is the assembly format printer/parser cannot infer all types with this format. If you want `$lhs`, `$rhs` and `$output` to always have the same type you can use a type variable to achieve this. Happy to show you how that's done!

Reply:
that would be great, thanks @AntonLydike

Reply:
Bit of an aside but doesn't `$operands` expand to something more like 
```
`(` $lhs `,` $rhs `)`
```
?